apps/backend/src/modules/categories/category.service.ts

```diff
@@ -112,6 +113,27 @@ class CategoryService implements Service {
 		return { items };
 	}
 
+	public async findCategories(
+		query: CategoriesGetRequestQueryDto,
+	): Promise<CategoriesGetAllResponseDto> {
+		if (query.categoryIds) {
+			const categories = await this.categoryRepository.findByIds(
+				JSON.parse(query.categoryIds) as number[],
+			);
+
+			const items = categories.map((categoryEntity) => {
+				const { createdAt, id, name, updatedAt } =
+					this.convertCategoryEntityToDto(categoryEntity);
+
+				return { createdAt, id, name, updatedAt };
+			});
+
+			return { items };
+		}
+
+		return await this.findAll();
+	}
```

Comment:
It would be more readable to inverse the condition
```
if (!query.categoryIds) {
return await this.findAll();
}
// here will be code moved from the if clauses
```